config/phpstan/static-reflection.neon

```diff
@@ -2,6 +2,7 @@ parameters:
     # see https://github.com/rectorphp/rector/issues/3490#issue-634342324
     featureToggles:
         disableRuntimeReflectionProvider: false
+        nodeConnectingVisitorCompatibility: false
```

Comment:
@TomasVotruba after various try, this disable seems the issue that create error in `vendor/rector/rector-symfony/config` that temporary not downgraded at PR:

- https://github.com/rectorphp/rector-src/pull/4859

Step to reproduce:

- uncomment `'vendor/rector/rector-symfony/config'` on build/config/config-downgrade.php
- Run command:

```
bin/rector process vendor/rector/rector-symfony/config/sets --config build/config/config-downgrade.php --ansi --dry-run --clear-cache --no-diffs
```

and it will got error:

```
 [ERROR] Could not process                                                                                              
         "/Users/samsonasik/www/rector-src/vendor/rector/rector-symfony/config/sets/twig/level/up-to-twig-127.php" file,
         due to:                                                                                                        
         "System error: "Trying to replace statement (Stmt_Return) with expression (Expr_MethodCall). Are you missing a 
         Stmt_Expression wrapper?"                                                                                      
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On line: 277
```

Then, try comment this `nodeConnectingVisitorCompatibility: false` flag:

```diff
-        nodeConnectingVisitorCompatibility: false
+#        nodeConnectingVisitorCompatibility: false
```

and it working ok:

```
➜  rector-src git:(main) ✗ bin/rector process vendor/rector/rector-symfony/config/sets --config build/config/config-downgrade.php --ansi --dry-run --clear-cache --no-diffs
 69/69 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
                                                                                                                        
 [OK] 6 files would have changed (dry-run) by Rector                                                                    
```

so the problem "parent" attribute seems somehow used in PHPStan internally probably on connecting scope or printing.

Reply:
Ok, it seems can be tweaked by early skip when `origNode` class is not equal with target node on RectifiedAnalyzer:

```diff
     private function isJustReprintedOverlappedTokenStart(Node $node, ?Node $originalNode): bool
     {
         if ($originalNode instanceof Node) {
-            return false;
+            return $originalNode::class === $node::class;
         }
```

Reply:
I created PR :

- https://github.com/rectorphp/rector-src/pull/4861

for it ;)